core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -346,4 +353,63 @@ public static void configureHadoopConf(Object maybeConfigurable, Object conf) {
 
     setConf.invoke(conf);
   }
+
+  /**
+   * Used to migrate tables from one catalog(source catalog) to another catalog(target catalog).
+   * Also, the table would be dropped off from the source catalog once the migration is successful.
+   *
+   * @param tableIdentifiers        a list of tableIdentifiers for the tables required to be migrated,
+   *                                if not specified all the tables would be migrated
+   * @param sourceCatalogProperties Source Catalog Properties
+   * @param targetCatalogProperties Target Catalog Properties
+   * @param sourceHadoopConfig      Source Catalog Hadoop Configuration
+   * @param targetHadoopConfig      Target Catalog Hadoop Configuration
+   * @return list of table identifiers for successfully migrated tables
+   */
+  public static List<TableIdentifier> migrateTables(List<TableIdentifier> tableIdentifiers,
+      Map<String, String> sourceCatalogProperties, Map<String, String> targetCatalogProperties,
+      Object sourceHadoopConfig, Object targetHadoopConfig) {
+    if (tableIdentifiers != null) {
+      tableIdentifiers.forEach(tableIdentifier -> Preconditions.checkArgument(
+          tableIdentifier != null, "Invalid identifier: %s", tableIdentifier));
+    }
+    Catalog sourceCatalog;
+    try {
+      sourceCatalog = loadCatalog(sourceCatalogProperties.get("catalogImpl"),
+          sourceCatalogProperties.get("catalogName"), sourceCatalogProperties, sourceHadoopConfig);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(String.format(
+          "Cannot initialize Source Catalog implementation %s: %s", sourceCatalogProperties.get("catalogImpl"),
+          e.getMessage()), e);
+    }
+    Catalog targetCatalog;
+    try {
+      targetCatalog = loadCatalog(targetCatalogProperties.get("catalogImpl"),
+          targetCatalogProperties.get("catalogName"), targetCatalogProperties, targetHadoopConfig);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(String.format(
+          "Cannot initialize Target Catalog implementation %s: %s", targetCatalogProperties.get("catalogImpl"),
+          e.getMessage()), e);
+    }
+    List<TableIdentifier> allIdentifiers = tableIdentifiers;
+    if (tableIdentifiers == null || tableIdentifiers.isEmpty()) {
+      List<Namespace> namespaces = (sourceCatalog instanceof SupportsNamespaces) ?
+          ((SupportsNamespaces) sourceCatalog).listNamespaces() : ImmutableList.of(Namespace.empty());
+      allIdentifiers = namespaces.stream().flatMap(ns ->
```

Comment:
I suspect this will run for a very long time, like when there a a lot of tables.